stix_shifter_modules/cybereason/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,471 @@
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+import re
+import json
+from datetime import datetime, timedelta
+from os import path
+import copy
+from stix_shifter_utils.utils import logger
+
+logger = logger.set_logger(__name__)
+
+START_STOP_PATTERN = r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z)"
+MAC = '^(([0-9a-fA-F]{2}[:-]){5}([0-9a-fA-F]{2}))$'
+EMAIL = r'^\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b$'
+CONFIG_MAP_PATH = "json/config_map.json"
+
+STOP_TIME = datetime.utcnow()
+
+
+class LinkNotFoundException(Exception):
+    pass
+
+
+class QueryStringPatternTranslator:
+    """
+    comparator values to match with supported data source operators
+    """
+    comparator_lookup = {
```

Comment:
@mdazam1942 does this now need to be extracted out into its own JSON file given the recent changes that went in or will it still work?

Reply:
@delliott90 it should work without a operators.json. we can include that after this pr is merged.

Reply:
looks like I am wrong. it needs operators.json. the base translator class looks for it

Reply:
@lakshmi51974368 so you need to create a file in `json/ `folder named `operators.json` and specify the operator mapping there, I already compiled the mapping for you. you can just copy paste this-

```
{
    "ComparisonExpressionOperators.And": "AND",  
    "ComparisonComparators.GreaterThan": "GreaterThan",
    "ComparisonComparators.GreaterThanOrEqual": "GreaterOrEqualsTo",
    "ComparisonComparators.LessThan": "LessThan",
    "ComparisonComparators.LessThanOrEqual": "LessOrEqualsTo",
    "ComparisonComparators.Equal": "Equals",
    "ComparisonComparators.NotEqual": "NotEquals",
    "ComparisonComparators.Like": "ContainsIgnoreCase",
    "ComparisonComparators.In": "Equals",
    "ComparisonComparators.Matches": "ContainsIgnoreCase",
    "ObservationOperators.And": "AND"
}

```

Reply:
The comparator lookup dictionary has been moved to operators.json file, and respective changes has been made to import it to query constructor as suggested.